### PR TITLE
remove is_training from cuDNN SDPA API

### PR DESCRIPTION
**Pull request by Cjkkkk**

### CODE REVIEW

**Comment by Cjkkkk** [comment]
@superbobry Hi, could you help review this? Also @kaixih from NVIDIA to help review this as well.

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 1]
```diff
@@ -978,14 +982,11 @@ def dot_product_attention(query: Array,
...
   check_layout(query, key, value, bias, q_seqlen, kv_seqlen, layout)
-  # check if flash attention is supported for this attention pattern
-  check_is_flash_attention(
-      query, key, layout, cudnn_version, bias is not None, is_training)
   if has_padding(mask_type) and (q_seqlen is None or kv_seqlen is None):
     raise ValueError("Require q_seqlen and kv_seqlen to generate padding mask")
   has_bias = bias is not None
   has_mask = mask is not None
-  has_dbias = has_bias and is_training and \
+  has_dbias = has_bias and \
```
Comment (kaixih) [inline jax/_src/cudnn/fused_attention_stablehlo.py:989]: The remove of `is_training` from here changes the behavior, right?

Also, I feel the better way to do this is not to pass the `variadic_args` around. Instead, we can compute it in `_dot_product_attention_fwd` and `_dot_product_attention_fwd_rule` respectively, where we should be able to access the `bias`, `mask`, `query` and `layout`, right?

Comment (Cjkkkk) [reply jax/_src/cudnn/fused_attention_stablehlo.py:989]: It does not change the behavior, the has_dbias is only used in bwd anyway and is_training is always True. Also bias/mask is set to one if it is None as Jax does not accept None here so it is better to compute before.

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 2]
```diff
@@ -963,7 +968,6 @@ def dot_product_attention(query: Array,
     dropout_rate: Dropout rate.
     qkv_layout: Layout string, with supported formats being BTNH, BNTH, BSNH,
                 BNSH.
-    is_training: choose to save activation or not.
```
Comment (superbobry) [inline jax/_src/cudnn/fused_attention_stablehlo.py:966]: AFAICT this API is not public, so removing a parameter is fine. Once it becomes public, we might need to decide on the API compatibility guarantees. See https://jax.readthedocs.io/en/latest/api_compatibility.html.

**Review by superbobry — APPROVED** [review]

**Comment by Cjkkkk** [comment]
@superbobry Could we merge this if there is nothing blocking?
